doc/source/user_guide/options.rst

```diff
@@ -430,6 +430,14 @@ display.html.use_mathjax                True         When True, Jupyter notebook
                                                      table contents using MathJax, rendering
                                                      mathematical expressions enclosed by the
                                                      dollar symbol.
+display.max_dir_items                   100          The number of columns from a dataframe that
+                                                     are added to dir. These columns can then be
+                                                     suggested by tab completion. 'None' value means
+                                                     unlimited.
+                                                     Due to caching changing the setting
```

Comment:
this 2nd part of the statement is very confusing. what do you mean.

Reply:
Is it the part about the caching that you find confusing? I will try to explain it better but it's a bit a technical detail so suggestions are welcome...

Here is an example of what it means. We have a dataframe with 5 columns 'a' to 'e'. Tab-completion knows about all columns, so `'d' in dir(df)` is True etc...

```
import pandas as pd
pd.get_option('display.max_dir_items')  # it is now 100, this is the current value and default
df = pd.DataFrame([], columns=list('abcde'))
'd' in dir(df)  # it knows that 'd' is a column, great!
```

Now change the `max_dir_items`  to 3. Then `dir` should know about `a`, `b` and `c` only, and not know about `d` and `e`. But it still does! This is because the result of dir is cached and the cache is still consired valid.

```
pd.set_option('display.max_dir_items', 3)
'd' in dir(df)   # it is still True !! 
```

I'm not sure exactly what invalidates the cache but for instance adding a column will do the trick. Suddenly it doesn't remember that 'd' is a column:
``` 
df['x'] = []  # add a column
'd' in dir(df)  # it now becomes False
```

Of course it will give the right result for new dataframes, ... 

``` 
df = pd.DataFrame([], columns=list('abcde'))
'a' in dir(df), 'd' in dir(df)  # (True, False) as expected
```

It is not a big deal I think but it could be confusing if it is undocumented. For instance, you are analysing a big dataset and are annoyed that `df.col<tab>` doesn't give you all columns that start with `col`.  So you type `pd.set_option('display.max_dir_items', None)` and then you try again. But it still wouldn't work. You basically have to make a copy first, add a column, ... basically do something that mean dir is evaluated again.

Reply:
ok this is not important as likely this option won't be interactively changed

remove the text about this